iceoryx_utils/include/iceoryx_utils/internal/relocatable_pointer/relative_ptr.hpp

```diff
@@ -13,291 +14,95 @@
+    /// @param[in] offset is the offset
+    /// @param[in] id is the unique id of the segment
+    RelativePointer(offset_t offset, id_t id) noexcept;
+
+    /// @brief constructs a RelativePointer pointing to the same pointee as ptr
+    /// @param[in] ptr the pointer whose pointee shall be the same for this
+    RelativePointer(ptr_t ptr = nullptr) noexcept;
+
+    /// @brief creates a RelativePointer from a BaseRelativePointer
+    /// @param[in] other is the BaseRelativePointer
+    RelativePointer(const BaseRelativePointer& other) noexcept;
+
+    /// @brief assign this to point to the same pointee as the BaseRelativePointer other
+    /// @param[in] other the pointer whose pointee shall be the same for this
+    /// @return reference to self
+    RelativePointer& operator=(const BaseRelativePointer& other) noexcept;
+
+    /// @brief assigns the RelativePointer to point to the same pointee as ptr
+    /// @param[in] ptr the pointer whose pointee shall be the same for this
+    /// @return reference to self
+    RelativePointer& operator=(ptr_t ptr) noexcept;
+
+    /// @brief dereferencing operator which returns a reference to the underlying object
+    /// @tparam U a template parameter to enable the dereferencing operator only for non-void T
+    /// @return a reference to the underlying object
     template <typename U = T>
-    typename std::enable_if<!std::is_void<U>::value, U&>::type operator*()
-    {
-        return *get();
-    }
+    typename std::enable_if<!std::is_void<U>::value, U&>::type operator*() noexcept;
 
-    T* operator->()
-    {
-        return get();
-    }
+    /// @brief access to the underlying object
+    /// @return a pointer to the underlying object
+    T* operator->() noexcept;
 
+    /// @brief dereferencing operator which returns a const reference to the underlying object
+    /// @tparam U a template parameter to enable the dereferencing operator only for non-void T
+    /// @return a const reference to the underlying object
     template <typename U = T>
-    typename std::enable_if<!std::is_void<U>::value, const U&>::type operator*() const
-    {
-        return *get();
-    }
+    typename std::enable_if<!std::is_void<U>::value, const U&>::type operator*() const noexcept;
 
-    T* operator->() const
-    {
-        return get();
-    }
+    /// @brief read-only access to the underlying object
+    /// @return a const pointer to the underlying object
+    T* operator->() const noexcept;
 
-    T* get() const
-    {
-        return static_cast<T*>(computeRawPtr());
-    }
+    /// @brief access the underlying object
+    /// @return a pointer to the underlying object
+    T* get() const noexcept;
 
-    operator T*() const
-    {
-        return get();
-    }
+    /// @brief converts the RelativePointer to a pointer of the type of the underlying object
+    /// @return a pointer of type T pointing to the underlying object
+    operator T*() const noexcept;
```

Comment:
added to issue

Reply:
This is debatable, as it constructs a new T*. You get call this function and it guarantees not to change the relative pointer itself. It probably should be const to not allow changing the data via the raw pointer then though. But there must be a const and non const version in this case since we need the possibility to get a non const pointer.
As it is now it is not critical to be "fixed".